analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -0,0 +1,316 @@
+                dependencies.add(it)
+            }
+        }
+        return dependencies.toSortedSet()
+    }
+
+    /**
+     * Extracts plain list of all packages
+     */
+    private fun extractPackages(node: List<JsonNode>, workingDir: File): Map<String, Package> {
+        val result = mutableMapOf<String, Package>()
+        val stack = Stack<JsonNode>()
+        stack.addAll(node)
+        while (!stack.empty()) {
+            val currentNode = stack.pop()
+            val pkg = extractPackage(currentNode, workingDir)
+            result["${pkg.id.name}:${pkg.id.version}"] = pkg
+        }
+        return result
+    }
+
+    /**
+     * Creates corresponding `Package` object from given `JsonNode` object
+     */
+    private fun extractPackage(node: JsonNode, workingDir: File) =
+        Package(
+            id = extractPackageId(node, workingDir),
+            declaredLicenses = extractDeclaredLicenses(node),
+            description = extractPackageField(node, workingDir, "description"),
+            homepageUrl = node["homepage"].textValueOrEmpty(),
+            binaryArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+            sourceArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+            vcs = VcsInfo(
+                type = VcsType.GIT,
+                url = node["url"].textValueOrEmpty(),
+                revision = node["revision"].textValueOrEmpty()
+            )
+        )
+
+    private fun runInspectRawField(pkgName: String, workingDir: File, field: String): String {
+        return run(workingDir, "inspect", pkgName, "--raw", field).stdout
+    }
+
+    private fun removeProjectPackage(rootNode: JsonNode, definitionFile: File): List<JsonNode> {
+        val projectPackage: JsonNode? = rootNode.find {
+            // Contains because conanfile.py's reference string often includes other data.
+            it["reference"].textValueOrEmpty().contains(definitionFile.name)
+        }
+        return if (projectPackage != null) {
+            rootNode.minusElement(projectPackage)
+        } else {
+            rootNode.toList<JsonNode>()
+        }
+    }
+
+    private fun extractDeclaredLicenses(node: JsonNode): SortedSet<String> =
+        sortedSetOf<String>().apply {
+            val license = node["license"]
+            license?.forEach {
+                add(it.textValueOrEmpty())
+            }
+        }
+
+    private fun extractPackageId(node: JsonNode, workingDir: File) =
+        Identifier(
+            type = "Conan",
+            namespace = "",
+            name = extractPackageField(node, workingDir, "name"),
+            version = extractPackageField(node, workingDir, "version")
+        )
+
+    /**
+     * Runs 'conan inspect --raw' over a specified package to extract specified field.
+     */
+    private fun extractPackageField(node: JsonNode, workingDir: File, field: String): String {
+        return runInspectRawField(node["display_name"].textValueOrEmpty(), workingDir, field)
```

Comment:
Again, does `textValueOrEmpty()` make sense here, i.e. does the code properly handle an empty string? If not, we should probably just use `textValue()`. The same is probably true for more places where `textValueOrEmpty()` is used.